cost-analyzer/templates/_helpers.tpl

```diff
@@ -25,13 +25,32 @@ Set important variables before starting main templates
 {{/*
 Kubecost 2.0 preconditions
 */}}
-{{ if .Values.federatedETL }}
-  {{ if .Values.federatedETL.primaryCluster }}
-    {{ fail "In Kubecost 2.0, there is no such thing as a federated primary. If you are a Federated ETL user, this setting has been removed. Make sure you have kubecostAggregator.deployMethod set to 'statefulset' and federatedETL.federatedCluster set to 'true'." }}
+{{ define "kubecostV2-preconditions" }}
+  {{/*Get all StatefulSets in this namespace. Iterate through them and fail if any include the name "aggregator"*/}}
+  {{ if and (semverCompare "<2.0.0-0" .Chart.Version)}}
```

Comment:
I need to double check that this `if` statement does what I expect. I want to check if the _current_ Chart version is pre-2.0. Not the version of the chart we're upgrading to.

Reply:
I think Helm attaches metadata about the current release to some object in the cluster, but I'm not sure in what form. This command has to be accessing _something_ in the cluster that should give you the current version in-cluster:
```
→ helm get metadata --namespace kubecost-cloudcost kubecost-cloudcost
NAME: kubecost-cloudcost
CHART: cost-analyzer
VERSION: 2.0.0
APP_VERSION: development
NAMESPACE: kubecost-cloudcost
REVISION: 42
STATUS: deployed
DEPLOYED_AT: 2024-01-24T10:40:35-08:00
```